tools/generator/src/XcodeProjContainerReference.swift

```diff
@@ -0,0 +1,29 @@
+import PathKit
+
+/// In XcodeProj, a `referencedContainer` in a `XCScheme.BuildableReference`
+/// accepts a string in the format `container:<path-to-xcodeproj-dir>`. This
+/// type represents that structure.
+public struct XcodeProjContainerReference: Equatable {
```

Comment:
It seems odd that we need a whole type for this. Just wondering why it's not a simple computed string property on another type, like `Project`.

Reply:
I debated this a bit. You will see in the next PR that this type is an argument to a function that creates the schemes.

```swift
    static func createXCSchemes(
        referencedContainer: XcodeProjContainerReference,
        pbxTargets: [TargetID: PBXNativeTarget]
    ) throws -> [XCScheme] {
    // ...
    }
```

I feel that it makes it clearer what is being passed along. Here is what it looked like originally:

```swift
    static func createXCSchemes(
        referencedContainer: String,
        pbxTargets: [TargetID: PBXNativeTarget]
    ) throws -> [XCScheme] {
    // ...
    }
```

Without some doc that could be easily missed, it is not clear what this value should be. I dug around the `XcodeProj` code looking for a factory or an explanation of what the format should be. In the end, all of the values that I found looked like `container:path/to/Foo.xcodeproj`. I defined the type so that it was clear what this value is.

Reply:
What about a version that had `Project` as the parameter and `Project` had an extension on it to return this computed property? This just seems a bit heavy, and (oddly in contrast to your desired reason for it existing) too indirect.

It just confused me when I saw it, and even more when I looked at the implementation. I'm not opposed to it, just really wondering if it's needed.

Reply:
I guess instead of project you would do `FilePathResolver` which has the path to the project as a property already.

Reply:
I removed `XcodeProjContainerReference` and added `containerReference` to `FilePathResolver`.